### PR TITLE
ci: Add clang-tidy job

### LINKED ISSUE #2971: Set up clang-tidy in CI
`clang-tidy` seems like a useful linter. C++ could really use all the linters it can get.

https://github.com/marketplace/actions/clang-tidy-review looks useful but it might
be nice to be able to run clang-tidy locally as well thru nix. Perhaps this action
is better for that: https://github.com/marketplace/actions/clang-tidy-action . ie.
run clang-tidy thru nix, get the output file, then use the action to post inline comments
on the PR.

Would also be nice to only clang-tidy on changed lines, much like git-clang-format.

### PR DESCRIPTION
**Pull request by danobi**
This PR makes the CI start running clang-tidy. We're starting small
and over time hopefully we enable more checks. Right now this job
is structured to run over the entire codebase. So try to pair new lints
with the fixes in the same PR in the future.

This closes #2971.

##### Checklist

- [ ] Language changes are updated in `man/adoc/bpftrace.adoc` and if needed in `docs/reference_guide.md`
- [ ] User-visible and non-trivial changes updated in `CHANGELOG.md`
- [ ] The new behaviour is covered by tests

### CODE REVIEW

**Comment by danobi** [comment]
I'm thinking that rather than get fancy with the GH action, we should slowly allow-list lints into `.clang-tidy.yml` paired with codemods. That way, codebase always stays clang-tidy clean and we can do a naive clang-tidy over all source files. And we can run that inside nix env so it's easy to repro locally.

A script we maintain that can both `-fix` and use in CI would be nice.

File: src/ast/irbuilderbpf.cpp [hunk 1]
```diff
@@ -1605,7 +1605,8 @@ Value *IRBuilderBPF::CreateStrcontains(Value *val1,
 
 CallInst *IRBuilderBPF::CreateGetNs(TimestampMode ts, const location &loc)
 {
-  libbpf::bpf_func_id fn;
+  // Random default value to silence compiler warning
+  libbpf::bpf_func_id fn = libbpf::BPF_FUNC_ktime_get_ns;
```
Comment (jordalgo) [inline src/ast/irbuilderbpf.cpp:1609]: soo glad this is being fixed-ish

File: src/bpf_assembler.h [hunk 2]
```diff
@@ -0,0 +1,168 @@
+// BPF macro assembler copied from bcc/libbpf.h
```
Comment (jordalgo) [inline src/bpf_assembler.h:1]: what is this file for?

**Review by jordalgo — APPROVED** [review]
🧹

File: src/bpf_assembler.h (hunk 2 above)
Comment (danobi) [reply src/bpf_assembler.h:1]: For the test programs in bpffeature. We use assembler there

**Pull request merged**